src/script/script.cpp

```diff
@@ -198,6 +198,13 @@ unsigned int CScript::GetSigOpCount(const CScript& scriptSig) const
     return subscript.GetSigOpCount(true);
 }
 
+bool CScript::IsPayToBasicStandardTemplate() const
```

Comment:
Minor bike shedding but "Basic Standard" and "Standard Template" are somewhat confusing terms. Are there non-standard templates?

Reply:
Yes... Perhaps "standard" and "basic" should be flipped terminology wise, but as is:

- Standard is the 32 byte argument form interpreted as the StandardTemplateHash defined in this PR
- Basic is the ScriptType that is just a script with a single <h:32> CTV. 

CTV as a NOP is extensible in the future to support different templates via having a different length arg + whatever rules around that you want.

BasicStandardTemplate is just for enabling relaying of a single special cased barescript type (for congestion control uses). TBH it *could* be dropped from this PR and handled separately, although it makes testing more painful.

Reply:
discussion of this moved here https://github.com/bitcoin/bips/pull/1118